angular/projects/admin-nrpti/src/app/mines/mines-records-list/mines-records-list.component.ts

```diff
@@ -103,7 +105,175 @@ export class MinesRecordsListComponent implements OnInit, OnDestroy {
+      FilterType.RadioPicker,
+      'LNG Published State',
+      new CheckOrRadioFilterDefinition([
+        new RadioOptionItem('lngPublishedState', 'Published', 'true'),
+        new RadioOptionItem('lngUnpubState', 'Unpublished', 'false')
+      ])
+    );
+
+    const responsibleAgencyFilter = new FilterObject(
+      'agency',
+      FilterType.MultiSelect,
+      'Responsible Agency',
+      new MultiSelectDefinition(Picklists.agencyPicklist.map(value => {
+        return { value: value, displayValue: value, selected: false, display: true };
+      }), 'Begin typing to filter agencies...', '')
+    );
+
+    const issuedUnderRegFilter = new FilterObject(
+      'regulation',
+      FilterType.MultiSelect,
+      'Issued Under which Regulation',
+      new MultiSelectDefinition(Picklists.getAllRegulations().map(value => {
+        return { value: value, displayValue: value, selected: false, display: true };
+      }), 'Begin typing to filter regulations...', '')
+    );
+
+    const sourceSystemFilter = new FilterObject(
+      'sourceSystemRef',
+      FilterType.Dropdown,
+      'Source System',
+      new DropdownDefinition(Picklists.sourceSystemRefPicklist)
+    );
+
+    const projectFilter = new FilterObject(
+      'projects',
+      FilterType.Checkbox,
+      'Project',
+      new CheckOrRadioFilterDefinition([
+        new OptionItem('lngCanada', 'LNG Canada'),
+        new OptionItem('coastalGaslink', 'Coastal Gaslink'),
+        new OptionItem('otherProjects', 'Other')],
+        true)
+    );
+
+    const documentsfilter = new FilterObject(
+      'hasDocuments',
+      FilterType.RadioPicker,
+      'Documents',
+      new CheckOrRadioFilterDefinition([
+        new RadioOptionItem('yesDoc', 'Yes', 'true'),
+        new RadioOptionItem('noDoc', 'No', 'false')
+      ])
+    );
+
+    this.filters = [
+      issuedDateFilter,
+      entityTypeFilter,
+      publishedStatefilter,
+      activityTypeFilter,
+      issuedUnderActFilter,
+      lngPublishedStatefilter,
+      responsibleAgencyFilter,
+      issuedUnderRegFilter,
+      sourceSystemFilter,
+      projectFilter,
+      documentsfilter
+    ];
+  }
+
+  executeSearch(searchPackage) {
+    console.log(searchPackage);
+    this.clearQueryParamsFilters();
+
+    // check keyword
+    if (searchPackage.keywords) {
+      this.queryParams['keywords'] = searchPackage.keywords;
+      // always change sortBy to '-score' if keyword search is directly triggered by user
+      if (searchPackage.keywordsChanged) {
+        this.tableData.sortBy = '-score';
+      }
+    }
+
+    // check subset
+    if (searchPackage.subset) {
+      this.queryParams['subset'] = [searchPackage.subset];
+    }
+
+    Object.keys(searchPackage.filters).forEach(filter => {
+      this.queryParams[filter] = searchPackage.filters[filter];
+    });
+
+    this.tableData.currentPage = 1;
+    this.submit();
+  }
+
+  private clearQueryParamsFilters() {
+    delete this.queryParams['keywords'];
```

Comment:
Are there other params that need to remain or can `this.queryParams` just be set to an empty object?

Reply:
Yeah, looks like in the current screens the queryParams holds some other values related to paging, etc. Originally I had it set to just do a `this.queryParams = {}` but that was causing issues, so this is a safety net for that. A better fix is to probably ensure queryParams is just for query params, and tableParams is handling all the table params without any crossover; checking into them now.